RickAndMorty-iOS/Views/Characters/CharacterDetailsViewController.swift

```diff
@@ -0,0 +1,253 @@
+        configureDataSource()
+        showEmptyData()
+        subscribeToViewModel()
+        updateTitleView()
+    }
+
+    func showEmptyData() {
+        snapshot.deleteAllItems()
+        snapshot.appendSections([.appearance, .info, .location, .episodes, .empty, .emptyInfo, .emptyLocation])
+        snapshot.appendItems(Array(repeatingExpression: EmptyData(id: UUID()), count: 1), toSection: .empty)
+        snapshot.appendItems(Array(repeatingExpression: EmptyData(id: UUID()), count: 3), toSection: .emptyInfo)
+        snapshot.appendItems(Array(repeatingExpression: EmptyData(id: UUID()), count: 2), toSection: .emptyLocation)
+        self.dataSource.apply(snapshot, animatingDifferences: true)
+    }
+
+    func subscribeToViewModel() {
+        viewModel.character.sink(receiveValue: { [self] characterInfo in
+            self.title = characterInfo.name
+            if !characterInfo.episode.isEmpty {
+                snapshot.deleteAllItems()
+                snapshot.appendSections([.appearance, .info, .location, .episodes])
+                snapshot.appendItems([CharacterDetails(characterInfo)], toSection: .appearance)
+                snapshot.appendItems([CharacterDetails(characterInfo), CharacterDetails(characterInfo), CharacterDetails(characterInfo)], toSection: .info)
+                snapshot.appendItems([CharacterDetails(characterInfo), CharacterDetails(characterInfo)], toSection: .location)
+                snapshot.appendItems(characterInfo.episode, toSection: .episodes)
+                self.dataSource.apply(snapshot, animatingDifferences: true)
+            }
+        }).store(in: &cancellables)
+    }
+}
+
+// MARK: - CollectionView DataSource
+extension CharacterDetailsViewController {
+    // swiftlint:disable cyclomatic_complexity
+    private func configureDataSource() {
+        dataSource = DataSource(collectionView: characterDetailsView.collectionView, cellProvider: { (collectionView, indexPath, characterInfo) -> UICollectionViewCell? in
+
+            let avatarCell = collectionView.dequeueReusableCell(withReuseIdentifier: AvatarCell.identifier, for: indexPath) as? AvatarCell
+            let infoCell = collectionView.dequeueReusableCell(withReuseIdentifier: InfoCell.identifier, for: indexPath) as? InfoCell
+
+            switch indexPath.section {
+            case 0:
+                hideLoadingAnimation(currentCell: avatarCell!)
+                if let character = characterInfo as? CharacterDetails {
+                    guard let image = character.item.image else { fatalError("Image not found") }
+                    self.avatarImageUrl = image
+                    avatarCell?.characterImage.sd_setImage(with: URL(string: image))
+                    return avatarCell!
+                }
+            case 1:
+                hideLoadingAnimation(currentCell: infoCell!)
+                if let character = characterInfo as? CharacterDetails {
+                    switch indexPath.item {
```

Comment:
label values, should be in a separate file as enum or static vars. Same as image names.